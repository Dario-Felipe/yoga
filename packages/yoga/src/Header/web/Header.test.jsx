import React from 'react';
import { screen, render } from '@testing-library/react';

import { ThemeProvider, Header } from '../..';

describe('<Header />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should have link', () => {
    const gympassLink = 'https://site.gympass.com/br';

    render(
      <ThemeProvider>
        <Header link={gympassLink} />
      </ThemeProvider>,
    );

    screen.getByRole('link', { href: gympassLink });
  });

  it('should show gympass logo', () => {
    render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>,
    );

    screen.getByTitle('Gympass Logo');
  });

  it('should show custom logo', () => {
    const CustomLogo = () => <div>Custom logo</div>;

    render(
      <ThemeProvider>
        <Header logo={CustomLogo} />
      </ThemeProvider>,
    );

    screen.getByText('Custom logo');
  });

  it('should show children', () => {
    render(
      <ThemeProvider>
        <Header>
          <p>Make wellbeing universal</p>
        </Header>
      </ThemeProvider>,
    );

    screen.getByText('Make wellbeing universal');
  });
});
