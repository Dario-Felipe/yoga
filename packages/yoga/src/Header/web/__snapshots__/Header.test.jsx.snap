// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c1 {
  margin-right: 20px;
  width: 80px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-shadow: 0 5px 5px -3px rgba(152,152,166,0.2),0 8px 10px 1px rgba(152,152,166,0.14),0 3px 14px 2px rgba(152,152,166,0.12);
  width: 100%;
  padding: 0 20px;
  height: 56px;
}

@media (min-width:1024px) {
  .c1 {
    width: 130px;
    margin-right: 48px;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 0 72px;
    height: 72px;
  }
}

<div>
  <header
    class="c0"
  >
    <svg
      class="c1"
    />
  </header>
</div>
`;
